README.md

```diff
@@ -0,0 +1,152 @@
+
+City id can be found here: http://bulk.openweathermap.org/sample/
+
+Example:
+curl http://127.0.0.1:8000/api/weather/id/833
+```
+
+## Exploring the Momento Cache Integration
+How to use Momento cache client and cache driver can be found in (WeatherController.php)[src/Controllers/WeatherController.php].
+
+Use Momento client directly:
+```php
+$authProvider = new EnvMomentoTokenProvider("MOMENTO_AUTH_TOKEN");
+$momentoClient = new SimpleCacheClient($authProvider, 60);
+$cacheName = "zipcode-cache";
+$momentoClient->createCache($cacheName);
+$apiKey = env("WEATHER_API_KEY");
+$url = "https://api.openweathermap.org/data/2.5/weather?q={$zipcode},{$countryCode}&appid={$apiKey}";
+$zipcodeWeatherInfo = "{$zipcode}-{$countryCode}";
+$result = $momentoClient->get($cacheName, $zipcodeWeatherInfo);
+if ($result->asHit()) {
+    return $result->asHit()->value();
+}
+elseif ($result->asMiss()) {
+    $res = $this->httpClient->get($url);
+    if ($res->getStatusCode() == 200) {
+        $json = $res->getBody();
+        // 10 minutes TTLc
+        $momentoClient->set($zipcodeWeatherInfo, $zipcodeWeatherInfo, $json, 600);
+        return $json;
+    }
+}
+```
+
+Use Momento as a Laravel cache driver:
+```php
+$apiKey = env("WEATHER_API_KEY");
+$url = "https://api.openweathermap.org/data/2.5/weather?q={$city}&appid={$apiKey}";
+$result = Cache::get($city);
+if (!is_null($result)) {
+    return $result;
+} else {
+    $res = $this->httpClient->get($url);
+    if ($res->getStatusCode() == 200) {
+        $json = $res->getBody();
+        // 10 minutes TTLc
+        Cache::put($city, $json, 600);
+        return $json;
+    }
+}
+```
+
+## Manual PHP Setup
+
+Need to install the following:
+
+- [PHP](https://www.php.net/manual/en/install.macosx.packages.php)
+- [Composer](https://getcomposer.org/doc/00-intro.md#installation-linux-unix-macos)
+- [gRPC for PHP](https://cloud.google.com/php/grpc)
+
+Add the repository and dependency to your project's `composer.json`:
+
+```json
+{
+  "repositories": [
+    {
+      "type": "vcs",
+      "url": "https://github.com/momentohq/laravel-example"
+    },
+    {
+      "type": "vcs",
+      "url": "https://github.com/momentohq/laravel-cache"
+    },
+    {
+      "type": "vcs",
+      "url": "https://github.com/momentohq/client-sdk-php"
+    }
+  ],
+  "require": {
+    "momentohq/laravel-example": "0.1.0",
+    "momentohq/laravel-cache": "0.1.2",
+    "momentohq/client-sdk-php": "0.2.0"
+  }
+}
+```
+
+Finally, add the required config to your `config/cache.php`:
+
+```php
+'default' => env('CACHE_DRIVER', 'momento'),
+
+'stores' => [
+        'momento' => [
+            'driver' => 'momento',
+            'cache_name' => env('MOMENTO_CACHE_NAME'),
+            'default_ttl' => 60,
+        ],
+],
+```
+
+Run `composer update` to install the necessary prerequisites.
+If your app is not able to automatically discover packages, run `php artisan package:discover` alternatively.
```

Comment:
Was this something you had to do?

Reply:
I thought this needs to be run to bind service providers from `laravel-cache` but this might be caused because I wasn't explicitly calling the repos that need to be referenced.
Once I added the repo property to composer.json, I was just able to do `composer update` so I'll remove this,